### PR TITLE
JDBC test cases

### PR DESCRIPTION
**Pull request by kruthar**
First shot at test cases for the JDBC module. This should probably have come before I started working a lot on the JDBC module, but, better late than never. 

Uses an in-memory HSQLDB instance which doesn't leave any artifacts behind. Tests the 5 major operations. There may be other edge cases, please let me know. Otherwise, I hope this is a good first step to test cases.

There are several TODOs. We cannot properly test correct field values until #132 has been fixed. Currently the tests only test proper counts. Once a fix for #132 hits, we can uncomment all the TODO lines.

### CODE REVIEW

File: jdbc/pom.xml [hunk 1]
```diff
@@ -41,5 +41,16 @@ LICENSE file.
...
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.12</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.hsqldb</groupId>
+      <artifactId>hsqldb</artifactId>
+      <version>2.3.3</version>
```
Comment (risdenk) [inline jdbc/pom.xml]: <scope>test</scope>

**Comment by kruthar** [comment]
test scoped the hsqldb dependency

File: jdbc/src/test/java/com/yahoo/ycsb/db/JdbcDBClientTest.java [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+                    );
+                }
+                createString.append(")");
+                jdbcConnection.prepareStatement(createString.toString()).execute();
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+            fail("Failed to prepare test");
+        }
+
```
Comment (risdenk) [inline jdbc/src/test/java/com/yahoo/ycsb/db/JdbcDBClientTest.java]: nit: extra line

File: jdbc/src/test/java/com/yahoo/ycsb/db/JdbcDBClientTest.java [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+            assertEquals("Assert third row key is user2", resultSet.getString(KEY_FIELD), "user2");
+            for (int i = 0; i < 3; i++) {
+                assertEquals("Assert third row fields contain preupdateString", resultSet.getString(FIELD_PREFIX + i), preupdateString);
+            }
+            resultSet.close();
+        } catch (SQLException e) {
+            e.printStackTrace();
+            fail("Failed updateTest");
+        }
+
```
Comment (risdenk) [inline jdbc/src/test/java/com/yahoo/ycsb/db/JdbcDBClientTest.java]: nit: extra line

File: jdbc/src/test/java/com/yahoo/ycsb/db/JdbcDBClientTest.java [hunk 4]
```diff
@@ -0,0 +1,317 @@
...
+    @Test
+    public void deleteTest() {
+        try {
+            String insertBeforeKey = "user0";
+            HashMap<String, ByteIterator> insertBeforeMap = insertRow(insertBeforeKey);
+            String deleteKey = "user1";
+            insertRow(deleteKey);
+            String insertAfterKey = "user2";
+            HashMap<String, ByteIterator> insertAfterMap = insertRow(insertAfterKey);
```
Comment (risdenk) [inline jdbc/src/test/java/com/yahoo/ycsb/db/JdbcDBClientTest.java]: not used?

File: jdbc/src/test/java/com/yahoo/ycsb/db/JdbcDBClientTest.java [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+            // TODO: Uncomment this assertEquals when the fix is made.
+            //assertEquals("Assert " + field + " was read correctly", insertMap.get(field), readResultMap.get(field));
+        }
+    }
+
+    @Test
+    public void deleteTest() {
+        try {
+            String insertBeforeKey = "user0";
+            HashMap<String, ByteIterator> insertBeforeMap = insertRow(insertBeforeKey);
```
Comment (risdenk) [inline jdbc/src/test/java/com/yahoo/ycsb/db/JdbcDBClientTest.java]: not used?

**Comment by kruthar** [comment]
Fixed the noted issues.

File: jdbc/src/test/java/com/yahoo/ycsb/db/JdbcDBClientTest.java [hunk 6]
```diff
@@ -0,0 +1,317 @@
+package com.yahoo.ycsb.db;
```
Comment (cmccoy) [inline jdbc/src/test/java/com/yahoo/ycsb/db/JdbcDBClientTest.java]: Could you add a license here?

**Comment by kruthar** [comment]
license added

**Comment by cmccoy** [comment]
Thanks - +1 to more tests. Would you mind squashing some of these commits? Otherwise looks good.

**Comment by kruthar** [comment]
@cmccoy - commits squashed
